valuable/src/pointer.rs

```diff
@@ -0,0 +1,196 @@
+//! Valuable value pointer.
```

Comment:
It might be nice if the docs here provided more of an explanation of what this is for? 

Based on just the module name, my initial assumption would have been that this is for recording actual pointer values (e.g. `*mut T`/`*const T`). So, it might be worth saying something about how this represents a path traversal to a field of a different `Valuable` value.

Maybe:
```suggestion
//! References to fields of [`Valuable`] values.
//!
//! A [`Pointer`] stores a path traversal to a particular value in a nested
//! [`Valuable`] structure. For example, a [`Pointer`] might refer to the field `z`
//! of the field `y` of the value `x`. [`Pointer`] paths can also include indices into
//! [`Listable`] and [`Tupleable`] values, in order to represent expressions like
//! `x.y[3].0`.
```

Reply:
(btw, the term "pointer" in this PR is originated from JSON pointer.)